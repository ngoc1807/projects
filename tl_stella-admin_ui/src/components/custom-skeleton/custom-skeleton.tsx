import { FC } from "react";
import { Skeleton, SxProps, Theme } from "@mui/material";

interface Props {
  isInTable?: boolean;
  sx?: SxProps<Theme>;
  [x: string]: unknown;
}

export const CustomSkeleton: FC<Props> = ({
  isInTable = false,
  sx,
  ...rest
}) => {
  return (
    <Skeleton
      variant="rectangular"
      sx={{
        marginLeft: isInTable ? "auto" : undefined,
        ...sx,
      }}
      {...rest}
    />
  );
};
