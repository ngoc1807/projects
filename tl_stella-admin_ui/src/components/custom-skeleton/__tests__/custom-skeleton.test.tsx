import { cleanup, render } from "@testing-library/react";
import { CustomSkeleton } from "../custom-skeleton";

describe("custom-skeleton", () => {
  afterEach(() => {
    cleanup();
  });
  it("should render correctly when only required props are passed", () => {
    const { container } = render(<CustomSkeleton height={64} width="100%" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render correctly when optional prop isInTable is passed as true", () => {
    const { container } = render(
      <CustomSkeleton
        height={64}
        isInTable={true}
        width="100%"
        sx={{ fontWeight: "bold" }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
