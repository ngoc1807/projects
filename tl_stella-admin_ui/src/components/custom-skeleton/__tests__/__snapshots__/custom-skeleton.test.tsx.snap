// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-skeleton should render correctly when only required props are passed 1`] = `
<span
  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
  style="width: 100%; height: 64px;"
/>
`;

exports[`custom-skeleton should render correctly when optional prop isInTable is passed as true 1`] = `
<span
  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-th62h7-MuiSkeleton-root"
  style="width: 100%; height: 64px;"
/>
`;
